subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -1195,6 +1199,29 @@ enum hawkbit_response hawkbit_probe(void)
 	return hb_context.code_status;
 }
 
+#ifdef CONFIG_HAWKBIT_AFTER_UPDATE_CUSTOM
+int hawkbit_set_after_update_cb(hawkbit_after_update_cb_handler_t cb)
+{
+	if (cb == NULL) {
+		LOG_ERR("Invalid callback");
+		return -EINVAL;
+	}
+
+	hawkbit_after_update_cb_handler = cb;
+
+	return 0;
+}
+#endif /* CONFIG_HAWKBIT_AFTER_UPDATE_CUSTOM */
+
+static void after_update_default_cb(void)
+{
+#ifdef CONFIG_HAWKBIT_AFTER_UPDATE_REBOOT
```

Comment:
you may use `if(IS_ENABLED(...))` construct so that this goes through the compiler and minimizes the risk of letting compilation errors slip through. Likely applies to a couple other places in the change set.